packages/core/src/util/disableInstrumentation.js

```diff
@@ -0,0 +1,113 @@
+/*
+ * (c) Copyright IBM Corp. 2025
+ */
+
+'use strict';
+
+const { DISABLABLE_INSTRUMENTATION_GROUPS } = require('../tracing/constants');
+
+/** @type {import('../util/normalizeConfig').InstanaConfig} */
+let config;
+
+/** @type {import('../util/normalizeConfig').AgentConfig} */
+let agentConfig;
+
+/**
+ * @param {import('../util/normalizeConfig').InstanaConfig} _config
+ */
+function init(_config) {
+  config = _config;
+}
+
+/**
+ * @param {import('../util/normalizeConfig').AgentConfig} _agentConfig
+ */
+function activate(_agentConfig) {
+  agentConfig = _agentConfig;
+}
+
+/**
+ * @param {string} instrumentationPath
+ */
+function getModuleName(instrumentationPath) {
+  // Extracts the module name from the instrumentationPath.
+  // Tries to match the pattern './instrumentation/<category>/<module>' and extract the <module> part.
+  // If that pattern doesn't match (e.g., in custom instrumentation cases),
+  // it falls back to extracting the last segment of the instrumentationPath after the final '/'.
+  const matchResult = instrumentationPath.match(/.\/instrumentation\/[^/]*\/(.*)/);
+  const moduleName = matchResult ? matchResult[1] : instrumentationPath.match(/\/([^/]+)$/)[1];
+  return moduleName.toLowerCase();
+}
+
+/**
+ * Extracts category and module from an instrumentation instrumentationPath
+ * @param {string} instrumentationPath
+ * @returns {{category: string, module: string}|null}
+ */
+function getCategoryAndModule(instrumentationPath) {
+  const match = instrumentationPath.match(/\.\/instrumentation\/([^/]+)\/([^/]+)/);
+  return match ? { category: match[1], module: match[2] } : null;
+}
+
+/**
+ * @param {*} cfg
+ * @param {object} options
+ * @param {string} [options.moduleName]
+ * @param {string} [options.instrumentationName]
+ * @param {string} [options.category]
+ */
+function shouldDisable(cfg, { moduleName, instrumentationName, category } = {}) {
+  const disableConfig = cfg.tracing?.disable;
```

Comment:
```suggestion
  const disableConfig = cfg?.tracing?.disable;
```

The config object may not include a `tracing` section. Or can we assume it? Anyway, a check is good IMO